### PR TITLE
🔧  Apply calico ClusterResourceSet to tilt and dev clusters

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Use ClusterResourceSets to apply the calico addons for tilt and dev clusters. This allows using local tilt resources to bring up fully functional clusters, without needing to apply CNI manually. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Apply calico ClusterResourceSet to tilt and dev clusters
```

### CODE REVIEW

File: templates/cluster-template.yaml
Comment (CecileRobertMichon) [inline templates/cluster-template.yaml:5]: @sedefsavas can you please confirm that adding this to the released cluster template won't have any negative effect on users who use it without cluster resource sets and continue to apply the CNI addon manually?

Comment (sedefsavas) [reply templates/cluster-template.yaml:5]: Yes, as long as there is no `ClusterResourceSet` object that has this label as selector, it will be a no-op.
`ClusterResourceSet` labels are user selected, so there is no special meaning of any of them.

**Comment by CecileRobertMichon** [comment]
/hold 

until #646 merges so I can add ipv6 in here as well

**Comment by CecileRobertMichon** [comment]
Added ipv6 

/hold cancel

**Review by devigned — APPROVED** [review]
/lgtm
/approve

Yay! The workload clusters will have networking by default!

Since @mboersma is out today and the changes aren't super contriversial, I'm just going to approve.

**Pull request merged**
